### PR TITLE
Quirky XML to JSON conversions

### PR DESCRIPTION
**Pull request by s3u**
There are many ways to convert xml to json. If the default xml2json does not work you, you can override it on table by table basis. To override, specify path to an xformers.json file. Here is an example.

```
{
     "some.table" : "modulename relative to process.cwd()"
}
```

### CODE REVIEW

**Comment by s3u** [comment]
DON'T MERGE YET.

**Comment by s3u** [comment]
Ready for review

**Comment by s3u** [comment]
My last comment should have said "Update CHANGES"

**Comment by s3u** [comment]
Please note - due to a local push before the second fix, this pull req ended up mixing npm info changes for the mon page (https://github.com/ql-io/ql.io/issues/306). Sorry about that.

File: modules/engine/lib/engine.js [hunk 1]
```diff
@@ -114,11 +115,37 @@ var Engine = module.exports = function(opts) {
...
+                            self.xformers[k] = require(v);
+                        }
+                        catch(e) {
+                            delete self.xformers[k];
+                            self.emitError('Unable to load xformer ' + v);
+                        }
+                    }
+                }
+                else {
+                    delete self.xformers[k];
```
Comment (prabhakhar) [inline modules/engine/lib/engine.js:138]: This delete is not needed as we haven't populated self.xformers[k].

Comment (s3u) [inline modules/engine/lib/engine.js:138]: It is, as the loop is mutating an object from line 122. If the module is not found, I delete it, warn and continue.

Comment (prabhakhar) [inline modules/engine/lib/engine.js:138]: ok. got it.

**Pull request merged**